File: dotnet/EntityFX.NetBenchamarks.Core/Scimark2/MonteCarlo.cs
namespace EntityFX.NetBenchmark.Core.Scimark2
{
    internal class MonteCarlo
    {
        const int SEED = 113;

        public static double num_flops(int Num_samples)
        {
            // 3 flops in x^2+y^2 and 1 flop in random routine

            return ((double)Num_samples) * 4.0;

        }



        public static double integrate(int Num_samples)
        {

            Random R = new Random(SEED);


            int under_curve = 0;
            for (int count = 0; count < Num_samples; count++)
            {
                double x = R.nextDouble();
                double y = R.nextDouble();

                if (x * x + y * y <= 1.0)
                    under_curve++;

            }

            return ((double)under_curve / Num_samples) * 4.0;
        }


    }
}

File: src/java/EntityFX/src/EntityFX/Core/Scimark2/MonteCarlo.java
package EntityFX.Core.Scimark2;

import java.util.Random;
import java.util.concurrent.ThreadLocalRandom;

public class MonteCarlo {

    private static final int sEED = 113;

    public static double num_flops(int num_samples) {
        return (((double)num_samples)) * 4.0;
    }

    public static double integrate(int num_samples) {
        Random R = ThreadLocalRandom.current();
        int under_curve = 0;
        for (int count = 0; count < num_samples; count++) {
            double x = R.nextDouble();
            double y = R.nextDouble();
            if (((x * x) + (y * y)) <= 1.0) 
                under_curve++;
        }
        return ((((double)under_curve) / ((double)num_samples))) * 4.0;
    }
    public MonteCarlo() {
    }
}
